hikari/impl/bot.py

```diff
@@ -990,3 +995,34 @@ async def murder(future: asyncio.Future[typing.Any]) -> None:
 
         _LOGGER.debug("closing event loop")
         loop.close()
+
+    @staticmethod
+    def _validate_activity(activity: undefined.UndefinedNoneOr[presences.Activity]) -> None:
```

Comment:
does this have to be on the class?

Reply:
yes, the stacklevel has to be hardcoded in the warnings to make the line of code it shows in the warning make any sense.